lib_xua/src/core/clocking/sw_pll_wrapper.xc

```diff
@@ -199,4 +212,12 @@ void restart_sigma_delta(chanend c_sw_pll, unsigned selected_mclk_rate)
     outct(c_sw_pll, XS1_CT_END);
 }
 
+void sw_pll_set_pll_to_nominal(chanend c_sw_pll, unsigned selected_mclk_rate)
```

Comment:
This is not an issue with this PR since it copies the previous scheme, but im not sure why we are doing 

outuint
outct
outuint
outct 

we only need to do
outuint
outuint
outct 

(and change the other side, of course)

Reply:
To make it actually safe we should actually do the following, where the two columns are the two tasks

outct     inct.      // sync
inct        outct

outuint.   inuint
outuint.   inuint

outct       inct     // close routes
inct.         outct

this is deadlock safe.

Reply:
hey @xross, your comment prompted me to look in the c_sw_pll channel use and I found an issue related to clockgen and sw_pll_task communicating incorrectly over the channel at initialisation time. I'll open an issue for any further work on improving the channel comms over the c_sw_pll channel.

Reply:
https://github.com/xmos/lib_xua/issues/512